iot/api/CloudIotTest/CloudIotTest.cs

```diff
@@ -0,0 +1,334 @@
+﻿// Copyright(c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using Google.Cloud.Iam.V1;
+using Google.Cloud.PubSub.V1;
+using Grpc.Core;
+using Xunit;
+using System;
+
+namespace GoogleCloudSamples
+{
+    // <summary>
+    /// Runs the sample app's methods and tests the outputs
+    // </summary>
+    public class CommonTests
+    {
+        private static readonly string s_projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        private static readonly string s_regionId = "us-central1";
+        private static readonly string s_serviceAccount = "serviceAccount:cloud-iot@system.gserviceaccount.com";
+
+
+        readonly CommandLineRunner _cloudIot = new CommandLineRunner()
+        {
+            VoidMain = CloudIotSample.Main,
+            Command = "CloudIotSample"
+        };
+
+        protected ConsoleOutput Run(params string[] args)
+        {
+            return _cloudIot.Run(args);
+        }
+
+        private readonly RetryRobot _retryRobot = new RetryRobot()
+        {
+            RetryWhenExceptions = new[] { typeof(Xunit.Sdk.XunitException) }
+        };
+
+        private void Eventually(Action action) => _retryRobot.Eventually(action);
+
+        private void CreatePubSubTopic(String name)
+        {
+            var publisher = PublisherClient.Create();

```

Comment:
Let's be consistent.
If you use "var" for static type specifier in line 53, you should also do that in line 54.

Reply:
Using var on line 53 and TopicName on line 54 doesn't strike me as difficult to read.  What are the benefits to consistently using typenames or var?

Reply:
When the static type of the variable is obvious from the expression, "var" is preferable IMO because:
  1. You write less code without sacrificing the readability
  2. If you later need to refactor the code, there will be less visible changes (text diffs)

This is in line with the [guidelines](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions) that are commonly used by MS around "var". (Search for "Implicitly Typed Local Variables" in the doc).

Reply:
Done.